### ISSUE TITLE
Not installing module at correct location with conda pip

### ISSUE THREAD

**Issue by tierriminator** [scikit-build/scikit-build#768]
This issue seems to be related with #434 and #370, but I think I am still missing something.
I have a conda environment setup (finch), where I installed cmake and gxx. My C++ project (zebra) uses Pybind11 with CMake.
So I followed the [basic example](https://scikit-build.readthedocs.io/en/latest/usage.html#example-of-setup-py-cmakelists-txt-and-pyproject-toml) from the documentation and am now trying to install the package with `pip install .`.
Everything is built and installed successfully, as reported by `pip`.
However when importing in Python, I get a `ModuleNotFoundError`.
Investigating further, I see that the package is not being installed in the site-packages directory (`~/miniconda3/envs/finch/lib/python3.10/site-packages`), but only the dist-info.
I can find the compiled target in `_skbuild/{platform}/cmake-install/zebra`, so this seems to work.
My `CMakeLists.txt` looks as follows:
```
project(zebra LANGUAGES CXX)
[...]
find_package(pybind11 REQUIRED)
[...]
pybind11_add_module(zebra 
    [...]
)

install(TARGETS zebra LIBRARY
  DESTINATION zebra
)
```

**Comment by henryiii** [comment]
Do you have a repo I can look at? I'm guessing it's not picking up the files because of a name mismatch or missing package data.

**Comment by tierriminator** [comment]
Yes, I do. The relevant part of it is here: https://github.com/MeteoSwiss-APN/finch/tree/setup/zebra

**Comment by tierriminator** [comment]
After digging around a bit, I found that the module is being installed in `~/miniconda3/envs/finch`, in other words, directly inside the root directory of the conda environment.

**Comment by tierriminator** [comment]
When running `python setup.py install`, the Python egg is actually installed in the correct location. So this seems to work for now as a workaround.

**Comment by henryiii** [comment]
I'm currently working though the install paths parts of scikit-build-core. Once I get this worked out, hopefully I'll have something to backport here. Your package looks reasonably simple, I could use that as an example. Quick question: where are you getting pybind11 from? I don't see it in your backend requirements.

**Comment by tierriminator** [comment]
Yes, the repo is very new and the C++ part (zebra) is currently quite small.
I'm getting pybind11 from conda-forge. I provide a `requirements.yml` in the main repo (finch) with all the dependencies for finch (and zebra).

**Comment by tierriminator** [comment]
After I ran into several problems when using the legacy `python setup.py install`, I took a closer look at the issue.
The main problem seems to be that the CMake artifact, which is the extension module I want to install, will be classified as  a data file (see `setuptools_wrap._classify_file`). This works well with Python eggs, since they package all data files together in the egg. However with `wheel` packages, data files which are not part of a package directory won't be installed in the final `wheel` [(reference)](https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html#). This means that top-level CMake-based extension modules won't be installed at the correct location with `pip`.

**Comment by tierriminator** [comment]
So I finally figured out the problem.
I was trying to install the extension module itself, without wrapping it inside a package. Installing a single python extension module without a package doesn't seem to be supported yet (correct me if I'm wrong).
So I created an `__init__.py` inside the `zebra` directory, installed the extension module with cmake into `zebra` as well and added `zebra` to the list of packages in `setup.py`. Now everything works well.
